common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1572,3 +1572,11 @@ def set_user_partitions_for_scheme(self, partitions, scheme):
             if p.scheme != scheme
         ]
         self.user_partitions = other_partitions + partitions  # pylint: disable=attribute-defined-outside-init
+
+    def is_discussion_enabled(self, user):
+        """
+        Return True if Discussion is enabled for a course; else False
+        """
+        if not any([tab.type == 'discussion' and tab.is_enabled(self, user) for tab in self.tabs]):
```

Comment:
@nasthagiri 
Do we include complete tabs info (i.e. `CourseTabsList`) from course to course_overview or just a new field as list of all enabled tabs would work too?
Technically, it would as in mobile_api views we would have discussion tab in that list to know if we return discussion api url
and for discussion api course validation we are already working with course and we can use its tabs.is_enabled method to determin whether to raise not found error or not.
Logically, I wanted to know whether we need other information for future access or not?